tests/fpga/auto_opt_fpga.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2019-2021 ETH Zurich and the DaCe authors. All rights reserved.
+""" FPGA Tests for Auto Optimization """
+
+import dace
+import numpy as np
+from dace.transformation.interstate import FPGATransformSDFG
+from dace.transformation.auto import fpga as aopt
+
+N = dace.symbol('N')
+
+
+def test_global_to_local(size: int):
+    '''
+    Tests global_to_local optimization
+    :return:
+    '''
+    @dace.program
+    def global_to_local(A: dace.float32[N], B: dace.float32[N]):
+        for i in range(N):
+            tmp = A[i]
+            B[i] = tmp + 1
+
+    A = np.random.rand(size).astype(np.float32)
+    B = np.random.rand(size).astype(np.float32)
+
+    sdfg = global_to_local.to_sdfg()
+    sdfg.apply_transformations([FPGATransformSDFG])
+
+    aopt.fpga_global_to_local(sdfg)
+
```

Comment:
I don't understand. shouldn't this be under `aopt.auto_optimize(sdfg, dace.DeviceType.FPGA)`?

Reply:
Sure, I can do that! But for the moment being I will just have those two specific autoopts, that are not applied for CPU/GPU

Reply:
And @tbennun , now that I think about it, the SDFG  must have been already Transformed for FPGA (otherwise we find less optimization opportunities). Is there a way to check this?

Reply:
we already added "apply_gpu_transformations()" to the case where devicetype == GPU, so why not here as well?

Reply:
as I said in our previous discussion, we do some transformations before the gpu transform AND after, so it's really convenient to have control over this @TizianoDeMatteis

Reply:
> we already added "apply_gpu_transformations()" to the case where devicetype == GPU, so why not here as well?

I can not see it into auto_optimize. Or is it up to the user to do this before actually invoking auto-optimizations with devicetype==GPU?

Reply:
sorry, we probably have not merged it yet: https://github.com/sancierra/dace/blob/autoopt-fusion/dace/transformation/auto_optimize.py#L423